kotest-common/src/jvmMain/kotlin/io/kotest/mpp/replay.kt

```diff
@@ -15,23 +15,15 @@ actual suspend fun replay(
          action(it)
       }
    } else {
-      val executor = Executors.newFixedThreadPool(threads, NamedThreadFactory("replay-%d"))
-      val error = AtomicReference<Throwable>(null)
-      for (k in 0 until times) {
-         executor.submit {
-            runBlocking {
-               try {
-                  action(k)
-               } catch (t: Throwable) {
-                  error.compareAndSet(null, t)
+      @OptIn(DelicateCoroutinesApi::class)
+      newFixedThreadPoolContext(threads, "replay").use { dispatcher ->
```

Comment:
What's the change here - just moving away from JVM executors to kotlin coroutine dispatchers?

Reply:
On the surface, yes. Beneath, the idea is to move testing as close as possible to common production conditions. kotlinx.coroutine dispatchers still use Executors under the hood, but there are subtle differences, e.g.

* the various Executor flavors schedule differently at their discretion (creating threads more or less eagerly),
* directly using `Executor.shutdown()` can trigger a BlockHound exception due to blocking synchronization,
* `newFixedThreadPoolContext` explicitly creates daemon threads.

The change is meant to be a safe default which avoids having successful tests, but then seeing the same code fail in production due to such subtle differences.